Realm/Realm/Handles/SessionHandle.cs

```diff
@@ -147,6 +171,26 @@ public void UnregisterProgressNotifier(ulong token)
             ex.ThrowIfNecessary();
         }
 
+        public void SubscribeNotifications(Session session)
+        {
+            Debug.Assert(!_notificationToken.HasValue, $"{nameof(_notificationToken)} must be null before subscribing.");
+
+            var managedSessionHandle = GCHandle.Alloc(session, GCHandleType.Weak);
```

Comment:
Technically, this is leaking the handle, but it seems there are a fair amount of other places where we do it. I filed https://github.com/realm/realm-dotnet/issues/2867 to fix it and we can probably leave this for whenever we fix them all.

Reply:
You're right. There were a few changes around this and apparently I overlooked free it.
Since you filed an issue for this I guess we can postpone it as you said.
In this specific case we should free the handle during `UnsubscribeNotifications`, which is also called during unbinding. I think that should be it, right?